core/base/abstractTriangulation/AbstractTriangulation.h

```diff
@@ -3647,15 +3730,19 @@ namespace ttk {
     // (neighboring) ranks (per MPI rank)
     std::vector<std::vector<SimplexId>> remoteGhostCells_{};
 
-    std::vector<SimplexId> edgeLidToGid_{};
+    std::vector<ttk::LongSimplexId> edgeLidToGid_{};
     std::unordered_map<SimplexId, SimplexId> edgeGidToLid_{};
-    std::vector<SimplexId> triangleLidToGid_{};
+    std::vector<ttk::LongSimplexId> triangleLidToGid_{};
```

Comment:
You should be able to keep the `SimplexId` type here if you (re-)template `exchangeGhostCells`.

Reply:
Is it a good idea to have an id type different for different simplices? That would require to cast all remaining ids when the domain changes when only simplices of smaller dimensions are kept. For example, when executing IntegraLines, the cells become the edges. `setGlobalIds` takes LongSimplexId types, so to add the array, it would needed to convert it to LongSimplexId. 
Is the cost of doing that smaller than the cost of storing the ids of intermediary simplices as LongSimplexId?

Reply:
Considering that the `IntegralLines` output should be a small part of the input dataset, I think that the memory gains of compressing the intermediate identifiers of the input dataset (and keeping the consistency with the rest of the Triangulation API) should outweigh the conversion cost.

Reply:
Understood, thank you for the feedback